eln/preference.js

```diff
@@ -0,0 +1,58 @@
+import API from 'src/util/api';
+import Roc from '../rest-on-couch/Roc';
+
+export async function preferencesFactory(id, options) {
+    const {
+        name = 'viewPreferences',
+        url = undefined,
+        database = 'eln',
+        initial = []
+    } = options;
+
+    const kind = 'viewPreferences';
+
+    var roc = new Roc({
+        url: url,
+        database: database,
+        kind: kind
+    });
+
+    var user = await roc.getUser();
+    id += '_' + user.username;
+
+    var existing = (await roc.view('entryByKindAndId', {key: [kind, id]}))[0];
+    var preferenceRoc;
+    var rocOptions = {
+        varName: id,
+        track: true
+    };
+    if (existing) {
+        preferenceRoc = await roc.document(existing._id, rocOptions);
+    } else {
+        var created = await roc.create({
+            $id: id,
+            $content: initial,
+            $kind: kind
+        });
+        preferenceRoc = await roc.document(created._id, rocOptions);
+    }
+
+    var preferenceVar = API.getVar(id);
+    API.setVariable(name, preferenceVar, ['$content']);
```

Comment:
I'd suggest using the `elnPreferences` variable instead. Sometimes it's handy to add first level properties that you know will not be saved to the database (because they are not in $content)